clustered/integration-test/src/test/java/org/ehcache/clustered/reconnect/ThrowingResiliencyStrategy.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.reconnect;
+
+import org.ehcache.Cache;
+import org.ehcache.clustered.client.internal.store.ReconnectInProgressException;
+import org.ehcache.spi.resilience.ResilienceStrategy;
+import org.ehcache.spi.resilience.StoreAccessException;
+
+import java.util.Map;
+
+//For tests
+public class ThrowingResiliencyStrategy<K, V> implements ResilienceStrategy<K, V> {
+  @Override
+  public V getFailure(K key, StoreAccessException e) {
+    throw new RuntimeException(e);
+  }
+
+  @Override
+  public boolean containsKeyFailure(K key, StoreAccessException e) {
+    throw new RuntimeException(e);
+  }
+
+  @Override
+  public void putFailure(K key, V value, StoreAccessException e) {
+    if (e.getCause() instanceof ReconnectInProgressException) {
```

Comment:
Why not throwing for other things? And something puzzles me. When is the `ReconnectInProgressException` thrown? I would expect it to be transparent to the user

Reply:
This is just for this test, not meant for user, The scope is limited for test which is only using put, and I want to only throw Reconnect exception, since some puts might also get timed out during the test run

Reply:
I know. But can you comment the code about the intent? The thing that feels weird is that all other methods are throwing the exception whatever it is.

BTW, I did something similar in `DuplicateTest` by using a dynamic proxy.